keel-web/src/test/kotlin/com/netflix/spinnaker/keel/rest/DeliveryConfigYamlParsingFilesTests.kt

```diff
@@ -0,0 +1,122 @@
+import com.netflix.spinnaker.keel.api.ec2.ApplicationLoadBalancerSpec
+import com.netflix.spinnaker.keel.api.ec2.ClassicLoadBalancerSpec
+import com.netflix.spinnaker.keel.api.ec2.ClusterSpec
+import com.netflix.spinnaker.keel.api.ec2.SecurityGroupSpec
+import com.netflix.spinnaker.keel.api.ec2.old.ClusterV1Spec
+import com.netflix.spinnaker.keel.api.titus.TestContainerVerification
+import com.netflix.spinnaker.keel.api.titus.TitusClusterSpec
+import com.netflix.spinnaker.keel.api.titus.TitusServerGroup.Location
+import com.netflix.spinnaker.keel.core.api.SubmittedDeliveryConfig
+import dev.minutest.junit.JUnit5Minutests
+import dev.minutest.rootContext
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.test.context.SpringBootTest
+import org.springframework.boot.test.context.SpringBootTest.WebEnvironment.NONE
+import strikt.api.expectCatching
+import strikt.assertions.isA
+import strikt.assertions.isEqualTo
+import strikt.assertions.isSuccess
+
+@SpringBootTest(webEnvironment = NONE)
+class DeliveryConfigYamlParsingFilesTests @Autowired constructor(
+
+  private val mapper: YAMLMapper
+
+): JUnit5Minutests {
+
+  fun tests() = rootContext<Unit> {
+    test("ec2 cluster") {
+      parseSuccessfully("cluster-example.yml")
+        .get { environments.first().resources.first().spec }
+        .isA<ClusterV1Spec>()
+    }
+
+    test("ec2 cluster with scaling policies") {
+      parseSuccessfully("ec2-cluster-with-autoscaling-example.yml")
+        .get { environments.first().resources.first().spec }
+        .isA<ClusterSpec>()
+    }
+
+    test("security group") {
+      parseSuccessfully("security-group-example.yml")
+        .get { environments.first().resources.first().spec }
+        .isA<SecurityGroupSpec>()
+    }
+
+    test("clb") {
+      parseSuccessfully("clb-example.yml")
+        .get { environments.first().resources.first().spec }
+        .isA<ClassicLoadBalancerSpec>()
+    }
+
+    test("alb") {
+      parseSuccessfully("alb-example.yml")
+        .get { environments.first().resources.first().spec }
+        .isA<ApplicationLoadBalancerSpec>()
+    }
+
+    test("titus cluster") {
+      parseSuccessfully("titus-cluster-example.yml")
+        .get { environments.first().resources.first().spec }
+        .isA<TitusClusterSpec>()
+    }
+
+    test("simple titus cluster") {
+      parseSuccessfully("simple-titus-cluster-example.yml")
+        .get { environments.first().resources.first().spec }
+        .isA<TitusClusterSpec>()
+    }
+
+    test("titus cluster with artifact") {
+      parseSuccessfully("titus-cluster-with-artifact-example.yml")
+        .get { environments.first().resources.first().spec }
+        .isA<TitusClusterSpec>()
+    }
+
+    test("titus cluster with test container") {
+      parseSuccessfully("titus-cluster-with-test-container.yml")
+        .get { environments.first().verifyWith.first() }
+        .isEqualTo(
```

Comment:
I think more detail here is actually better, but it diverges from the pattern above of checking just the type. Should we stick with it?

Reply:
I'd like to keep the additional detail, despite the inconsistency, since I'd ultimately like to have more checks on the other tests too.